clients/src/main/java/org/apache/kafka/clients/ClientResponse.java

```diff
@@ -51,7 +51,7 @@ public ClientRequest request() {
         return request;
     }
 
-    public Struct responseBody() {
+    public AbstractResponse responseBody() {
```

Comment:
A somewhat unfortunate aspect of this is that we now have casts (and sometimes unchecked ones) in various places. Assuming that making `ClientResponse` generic would not help, we could accept a `Class` as a parameter to `responseBody`, verify if the cast is possible and throw an exception with a good error message if not.